src/include/nodes/plannodes.h

```diff
@@ -44,7 +44,8 @@ typedef enum PlanGenerator
 typedef enum DMLAction
 {
 	DML_DELETE,
-	DML_INSERT
+	DML_INSERT,
+	DML_INVALID
```

Comment:
I do not think it's a good idea.

Reply:
we can get away from this if we change the order in if and check for updates first. something like that. what do you think?
```c
if (!AttributeNumberIsValid(plannode->actionColIdx))
{
...
}
else if (DML_INSERT == action)
{
...
}
else /* delete */
{
...
}
```

Reply:
> what do you think?

https://github.com/arenadata/gpdb/pull/609#discussion_r1438077428

Reply:
Fine. but won't this slow down the insertion and deletion operations? since at least the insertion will probably happen more often?

Reply:
This is exactly what was done in the 7X.

Reply:
in 7X, this is done only for Update whereas we will do it for all DML operations.

Reply:
How about
```c
if (DML_INSERT == action)
{
...
}
else if (DML_DELETE == action)
{
...
}
else
{
Assert(!AttributeNumberIsValid(plannode->actionColIdx))
...
}
```

Reply:
or
```c
switch (action)
{
case DML_INSERT: ... break;
case DML_DELETE: ... break;
default: Assert(!AttributeNumberIsValid(plannode->actionColIdx)); ... break;
}
```

Reply:
this does not solve the problem. since we still need to assign some value to the `action` variable, otherwise we will get an undefined behavior.

Reply:
Yes, we assign -1 before

Reply:
-1 is not a valid DMLAction value. or are you suggesting changing the variable type to int?

Reply:
> are you suggesting changing the variable type to int?

Yes, sure!